libres/lib/job_queue/lsf_driver.cpp

```diff
@@ -959,24 +927,61 @@ void lsf_driver_free_job(void *__job) {
     lsf_job_free(job);
 }
 
+namespace lsf {
+std::vector<std::string> split_string(std::string str, std::string delimiter) {
+    std::vector<std::string> splitted = {};
+    size_t pos = 0, next_pos = 0;
+
+    while ((next_pos = str.find(delimiter, pos)) != std::string::npos) {
+        splitted.push_back(str.substr(pos, next_pos - pos));
+        pos = next_pos + delimiter.size();
+    }
+    splitted.push_back(str.substr(pos, str.size() - pos));
+    return splitted;
+}
+
+/*
+ * Parses the given file containing colon separated hostnames, ie.
+ * "hname1:hname2:hname3". This is the same format as
+ * written by lsf_job_write_bjobs_to_file().
+ */
+std::vector<std::string> parse_hostnames(const char *fname) {
+    std::ifstream stream(fname);
+
+    std::string line;
+    if (std::getline(stream, line)) { //just read the first line
+        // bjobs uses : as delimiter
+        std::vector<std::string> hosts_w_star = split_string(line, ":");
+
+        // change "4*be-lsf01" to "be-lsf01"
+        std::vector<std::string> hosts;
+        std::transform(hosts_w_star.begin(), hosts_w_star.end(),
+                       std::back_inserter(hosts),
+                       [](const std::string &host) -> std::string {
+                           return split_string(host, "*").back();
```

Comment:
Python anti-pattern (imo). Use `std::basic_string::rfind` instead? Or eg add as a utility function:

```c++
std::string_view back_element(std::string_view str, char delimiter) {
    auto pos = str.rfind(delimiter);
    return pos == str.npos ? str : str.substr(pos);
}
```

Reply:
Maybe, I was trying to recreate existing implementation more cleanly here.